presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -277,74 +263,19 @@ public PlanWithProperties visitAggregation(AggregationNode node, StreamPreferred
                     .values().stream()
                     .map(functionRegistry::getAggregateFunctionImplementation)
                     .allMatch(InternalAggregationFunction::isDecomposable);
-            if (decomposable && node.getStep() == Step.SINGLE && !requiredProperties.isParallelPreferred()) {
+            if (decomposable && !requiredProperties.isParallelPreferred()) {
```

Comment:
I think you can get rid of this whole if-clause since we are not dealing with decomposition here.